src/main/scala/com/softwaremill/realworld/Endpoints.scala

```diff
@@ -9,34 +9,15 @@ import sttp.tapir.ztapir.ZServerEndpoint
 import zio.ZIO
 import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}
 
+// TODO add authentication
+// TODO format errors properly
 object Endpoints:
   case class User(name: String) extends AnyVal
-  val helloEndpoint: PublicEndpoint[User, Unit, String, Any] = endpoint.get
+
+  private val helloEndpoint: PublicEndpoint[User, Unit, String, Any] = endpoint.get
     .in("hello")
     .in(query[User]("name"))
     .out(stringBody)
   val helloServerEndpoint: ZServerEndpoint[Any, Any] = helloEndpoint.serverLogicSuccess(user => ZIO.succeed(s"Hello ${user.name}"))
 
-  given authorZioEncoder: zio.json.JsonEncoder[Author] = DeriveJsonEncoder.gen[Author]
-  given authorZioDecoder: zio.json.JsonDecoder[Author] = DeriveJsonDecoder.gen[Author]
-  given bookZioEncoder: zio.json.JsonEncoder[Book] = DeriveJsonEncoder.gen[Book]
-  given bookZioDecoder: zio.json.JsonDecoder[Book] = DeriveJsonDecoder.gen[Book]
-  val booksListing: PublicEndpoint[Unit, Unit, List[Book], Any] = endpoint.get
-    .in("books" / "list" / "all")
-    .out(jsonBody[List[Book]])
-  val booksListingServerEndpoint: ZServerEndpoint[Any, Any] = booksListing.serverLogicSuccess(_ => ZIO.succeed(Library.books))
-
-  val apiEndpoints: List[ZServerEndpoint[Any, Any]] = List(helloServerEndpoint, booksListingServerEndpoint)
-
-  val all: List[ZServerEndpoint[Any, Any]] = apiEndpoints
-
-object Library:
-  case class Author(name: String)
-  case class Book(title: String, year: Int, author: Author)
-
-  val books = List(
-    Book("The Sorrows of Young Werther", 1774, Author("Johann Wolfgang von Goethe")),
-    Book("On the Niemen", 1888, Author("Eliza Orzeszkowa")),
-    Book("The Art of Computer Programming", 1968, Author("Donald Knuth")),
-    Book("Pharaoh", 1897, Author("Boleslaw Prus"))
-  )
+  val all: List[ZServerEndpoint[Any, Any]] = List(ArticlesEndpoints.endpoints, List(helloServerEndpoint)).flatten
```

Comment:
adopt-tapir offers the possibility to generate `docs` endpoint automatically - the benefit is that one can use the generated `/docs` endpoint to easily navigate through the API. Can be added as follow-up :) WDYT?